catalogs/catalog-lakehouse-paimon/src/main/java/com/datastrato/gravitino/catalog/lakehouse/paimon/GravitinoPaimonColumn.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.catalog.lakehouse.paimon;
+
+import static com.datastrato.gravitino.catalog.lakehouse.paimon.utils.TypeUtils.fromPaimonType;
+import static com.datastrato.gravitino.catalog.lakehouse.paimon.utils.TypeUtils.toPaimonType;
+
+import com.datastrato.gravitino.connector.BaseColumn;
+import com.datastrato.gravitino.rel.Column;
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.EqualsAndHashCode;
+import org.apache.paimon.types.DataField;
+import org.apache.paimon.types.DataType;
+import org.apache.paimon.types.RowType;
+
+/** Implementation of {@link Column} that represents a column in the Paimon column. */
+@EqualsAndHashCode(callSuper = true)
+public class GravitinoPaimonColumn extends BaseColumn {
+
+  private GravitinoPaimonColumn() {}
+
+  /**
+   * Converts {@link GravitinoPaimonColumn} instance to inner column.
+   *
+   * @param id The id of inner column.
+   * @return The converted inner column.
+   */
+  public static DataField toPaimonColumn(int id, Column gravitinoColumn) {
+    DataType paimonType = toPaimonType(gravitinoColumn.dataType());
+    DataType paimonTypeWithNullable =
+        gravitinoColumn.nullable() ? paimonType.nullable() : paimonType.notNull();
+    return new DataField(
+        id, gravitinoColumn.name(), paimonTypeWithNullable, gravitinoColumn.comment());
+  }
+
+  /**
+   * Creates new {@link GravitinoPaimonColumn} instance from inner columns.
+   *
+   * @param rowType The {@link RowType} instance of inner column.
+   * @return New {@link GravitinoPaimonColumn} instances.
+   */
+  public static List<GravitinoPaimonColumn> fromPaimonColumn(RowType rowType) {
```

Comment:
seems RowType is not a PaimonColumn, so we should use a better name

Reply:
updated.